mono/metadata/icall.c

```diff
@@ -7391,6 +7391,28 @@ ves_icall_System_Environment_Exit (int result)
 	exit (result);
 }
 
+void
+ves_icall_System_Environment_FailFast (MonoStringHandle message, MonoExceptionHandle exception, MonoStringHandle errorSource, MonoError *error)
+{
+	if (MONO_HANDLE_IS_NULL (message)) {
+		g_warning ("CLR: Managed code called FailFast without specifying a reason.");
+	} else {
+		char *msg = mono_string_handle_to_utf8 (message, error);
+		g_warning ("CLR: Managed code called FailFast, saying \"%s\"", msg);
+		g_free (msg);
+	}
+
+	if (!MONO_HANDLE_IS_NULL (exception)) {
+		mono_print_unhandled_exception_internal ((MonoObject *) MONO_HANDLE_RAW (exception));
+	}
+
+	// NOTE: While this does trigger WER on Windows it doesn't quite provide all the
+	// information in the error dump that CoreCLR would. On Windows 7+ we should call
+	// RaiseFailFastException directly instead of relying on the C runtime doing it
+	// for us and pass it as much information as possible.
+	abort ();
```

Comment:
Windows 7 is the minimum requirement for a while now. Feel free to depend on it.
Notice that importing code from CoreFX was making it so.
And Vista is reall quite old now.

Reply:
If we are going to mention things, mention Windows 8 __failfast. Maybe another year or two down the line that'll be it.

Reply:
I knew it was minimum requirement for .NET Core but I was unsure about Mono. I wrote the relevant part in managed code but I need to translate it back to native code and test the dumps.

Reply:
Almost a year ago.
https://github.com/mono/mono/pull/9568